enclave/src/tls_ra.rs

```diff
@@ -107,7 +109,9 @@ pub unsafe extern "C" fn run_key_provisioning_server(
 ) -> sgx_status_t {
     let _ = backtrace::enable_backtrace("enclave.signed.so", PrintFormat::Short);
 
-    let cfg = match tls_server_config(sign_type) {
+    let ocall_api = OCallComponentFactory::get_attestation_api();
+
+    let cfg = match tls_server_config(sign_type, ocall_api) {
         Ok(cfg) => cfg,
```

Comment:
At the top-level entry points (e-call functions), we create an ocall_api object and pass it down to the functions it calls, allowing those functions to be more generic and therefore also better testable.